File: #0442.find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {

        vector<int> result;
        for (int num_ : nums) {
            int num = std::abs(num_);
            nums[num - 1] *= -1;
            if (nums[num - 1] > 0) {
                result.push_back(num);
            }
        }
        return result;
    }
};

File: #0442.find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    const result = []; nums.unshift(0);
    for (let i = 0; i < nums.length; i++) {
        const index = Math.abs(nums[i]);
        if (nums[index] < 0) {
            result.push(index);
        }
        nums[index] *= -1;
    }
    return result;  
};
